From d3f531de2fbe11fa77c3e10951a6cdef890784f7 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Fri, 19 Oct 2018 16:53:25 +0800
Subject: [PATCH 4900/5242] MLK-19993 ARM64: dts: imx8qm-mek: move the mipi
 csi en/rst to specific node

commit  64f3b21f4ff81dde54ce18e7745affbd726a9520 from
https://source.codeaurora.org/external/imx/linux-imx.git

Move the mipi csi en/rst pinctrl configuration from gpio1 to
specific node for mipi csi.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi |   20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index 72ccfc8..f475c42 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -604,15 +604,19 @@
 			>;
 		};
 
-		pinctrl_gpio1: mipi_csi_en_rst{
+		pinctrl_mipi_csi0_en_rst: mipi_csi0_en_rst {
 			fsl,pins = <
 				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO1_IO27		0x00000021
 				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO1_IO28		0x00000021
-				SC_P_MIPI_CSI1_GPIO0_00_LSIO_GPIO1_IO30		0x00000021
-				SC_P_MIPI_CSI1_GPIO0_01_LSIO_GPIO1_IO31		0x00000021
 			>;
-		};
+                };
 
+		pinctrl_mipi_csi1_en_rst: mipi_csi1_en_rst {
+			fsl,pins = <
+				SC_P_MIPI_CSI1_GPIO0_00_LSIO_GPIO1_IO30         0x00000021
+				SC_P_MIPI_CSI1_GPIO0_01_LSIO_GPIO1_IO31         0x00000021
+			>;
+                };
 	};
 };
 
@@ -1154,21 +1158,20 @@
 };
 
 &gpio1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpio1>;
 	status = "okay";
 };
 
 &i2c0_mipi_csi0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	pinctrl-names = "default";
 	clock-frequency = <100000>;
 	status = "okay";
 
 	max9286_mipi@6A	 {
 		compatible = "maxim,max9286_mipi";
 		reg = <0x6A>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_mipi_csi0_en_rst>;
 		clocks = <&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "capture_mclk";
 		mclk = <27000000>;
@@ -1187,13 +1190,14 @@
 &i2c0_mipi_csi1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	pinctrl-names = "default";
 	clock-frequency = <100000>;
 	status = "disabled";
 
 	max9286_mipi@6A	 {
 		compatible = "maxim,max9286_mipi";
 		reg = <0x6A>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_mipi_csi1_en_rst>;
 		clocks = <&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "capture_mclk";
 		mclk = <27000000>;
-- 
1.7.9.5

